src/main/java/appeng/util/InventoryAdaptor.java

```diff
@@ -49,60 +41,29 @@
  */
 public abstract class InventoryAdaptor implements Iterable<ItemSlot>
 {
-
-	// returns an appropriate adaptor, or null
-	public static InventoryAdaptor getAdaptor( final Object te, final EnumFacing d )
+	public static InventoryAdaptor getAdaptor( final TileEntity te, final EnumFacing d )
 	{
-		if( te == null )
-		{
-			return null;
-		}
-
-		if( te instanceof ICapabilityProvider )
+		if( te != null )
 		{
-			ICapabilityProvider capProvider = (ICapabilityProvider) te;
-
-			// Attempt getting an IItemHandler for the given side via caps
-			IItemHandler itemHandler = capProvider.getCapability( CapabilityItemHandler.ITEM_HANDLER_CAPABILITY, d );
-			if( itemHandler != null && itemHandler.getSlots() > 0 )
+			if( te.hasCapability( CapabilityItemHandler.ITEM_HANDLER_CAPABILITY, d ) )
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1066)